PWGHF/HFC/Tasks/femtoDreamPairTaskCharmHadrons.cxx

```diff
@@ -9,357 +9,504 @@
+#include "Common/DataModel/Multiplicity.h"
+#include "Common/DataModel/Centrality.h"
+
+#include "DataFormatsParameters/GRPMagField.h"
+#include "DataFormatsParameters/GRPObject.h"
+#include "DetectorsBase/Propagator.h"
+
+#include "Framework/ASoAHelpers.h"
 #include "Framework/AnalysisTask.h"
-#include "Framework/runDataProcessing.h"
 #include "Framework/HistogramRegistry.h"
-#include "Framework/ASoAHelpers.h"
-#include "Framework/RunningWorkflowInfo.h"
-#include "Framework/StepTHn.h"
-
-#include "PWGCF/DataModel/FemtoDerived.h"
-#include "PWGCF/FemtoDream/Core/femtoDreamParticleHisto.h"
-#include "PWGCF/FemtoDream/Core/femtoDreamEventHisto.h"
-#include "PWGCF/FemtoDream/Core/femtoDreamPairCleaner.h"
-#include "PWGCF/FemtoDream/Core/femtoDreamContainer.h"
-#include "PWGCF/FemtoDream/Core/femtoDreamDetaDphiStar.h"
+#include "Framework/runDataProcessing.h"
+
+#include "PWGCF/FemtoDream/Core/femtoDreamCollisionSelection.h"
+#include "PWGCF/FemtoDream/Core/femtoDreamTrackSelection.h"
 #include "PWGCF/FemtoDream/Core/femtoDreamUtils.h"
 
+#include "PWGHF/Core/HfHelper.h"
+#include "PWGHF/DataModel/CandidateReconstructionTables.h"
+#include "PWGHF/DataModel/CandidateSelectionTables.h"
+#include "PWGHF/Utils/utilsBfieldCCDB.h"
+
 using namespace o2;
-using namespace o2::aod;
-using namespace o2::soa;
 using namespace o2::framework;
 using namespace o2::framework::expressions;
 using namespace o2::analysis::femtoDream;
 
-struct femtoDreamPairTaskCharmHadrons {
-
-  enum PairCharge {
-    positiveCharge = 1,
-    negativeCharge = -1
-  };
-
-  /// General options
-  Configurable<bool> ConfOptCPRPlotPerRadii{"ConfOptCPRPlotPerRadii", false, "Plot CPR per radii"};
-  Configurable<bool> ConfOptUseCPR{"ConfOptUseCPR", false, "Close Pair Rejection"};
-  Configurable<float> ConfOptCPRdeltaEtaMax{"ConfOptCPRdeltaEtaMax", 0.01, "Max. Delta Eta for Close Pair Rejection"};
-  Configurable<float> ConfOptCPRdeltaPhiMax{"ConfOptCPRdeltaPhiMax", 0.01, "Max. Delta Phi for Close Pair Rejection"};
-  Configurable<bool> ConfOptExtendedPlots{"ConfOptExtendedPlots", false, "Enable additional three dimensional histogramms. High memory consumption. Use for debugging"};
-  Configurable<float> ConfOptHighkstarCut{"ConfOptHighkstarCut", 100000., "Set a cut for high k*, above which the pairs are rejected"};
-  Configurable<bool> ConfOptisMc{"ConfOptisMc", false, "Enable additional Histogramms in the case of a MonteCarlo Run"};
-  Configurable<bool> ConfOptsmearingByOrigin{"ConfOptsmearingByOrigin", false, "Obtain the smearing matrix differential in the MC origin of particle 1 and particle 2. High memory consumption. Use with care!"};
-  Configurable<bool> ConfOptUse4D{"ConfOptUse4D", false, "Enable four dimensional histogramms (to be used only for analysis with high statistics): k* vs multiplicity vs multiplicity percentil vs mT"};
-  ConfigurableAxis ConfOptDummy{"ConfOptDummy", {1, 0, 1}, "Dummy axis"};
+struct femtoDreamProducer {
```

Comment:
Why are you replacing `femtoDreamPairTaskCharmHadrons` with `femtoDreamProducer` which already exists in another file?

Reply:
@vkucera sorry, it was by mistake, my .clang-format file is not not working correctly, due to that I was trying to format manually according to clang-format and this mistake happed. Though I corrected it now

Reply:
You don't need to format anything. That's why we have automatic formatting pull requests, such as the one linked here. Please read the contribution guidelines on the documentation page.

Reply:
I already read about automatic formatting, but I waited a few moments and nothing happened, so I tried manually. Thank you for mentioning it. I will keep this in my mind

Reply:
The guidelines tell you:

> You have to merge this formatting PR to update your branch (and your opened PR).

Reply:
I've attempted to correct it according to the naming convention. Please let me know if I missed anything or if you notice any glaring issues that need correction immediately. Minor adjustments can be considered for the next commit